### PR TITLE
media: Return bool when DataSource.close()

### PR DESCRIPTION
**Pull request by junmin-kim**
DataSource.close returns success or not

### CODE REVIEW

**Comment by junmin-kim** [comment]
@jsdosa @jarvisjang Should we support return value of close? If not, we should modify Doxygen comment

**Comment by seinfra** [comment]
Target : [12bad6051e80c93476fe209ea4388f06731b6128] - Code Rule Check Result:
framework/include/media/DataSource.h:66: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/DataSource.h:66: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/FileInputDataSource.h:33: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/include/media/FileInputDataSource.h:54: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/FileInputDataSource.h:54: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:60: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:66: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:66: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/FileInputDataSource.h:66: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:113: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/FileOutputDataSource.h:32: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/include/media/FileOutputDataSource.h:48: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/FileOutputDataSource.h:48: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileOutputDataSource.h:58: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/FileOutputDataSource.h:58: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileOutputDataSource.h:64: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileOutputDataSource.h:70: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileOutputDataSource.h:70: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/FileOutputDataSource.h:70: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileOutputDataSource.h:111: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)

**Comment by jsdosa** [comment]
@junmin-kim  please should modify comment for return type

File: framework/src/media/FileOutputDataSource.cpp [hunk 1]
```diff
@@ -54,12 +54,14 @@ bool FileOutputDataSource::open()
 	return false;
 }
 
-void FileOutputDataSource::close()
+bool FileOutputDataSource::close()
 {
-	if (mFp) {
-		fclose(mFp);
+	if (mFp && fclose(mFp) == 0) {
```
Comment (Taejun-Kwon) [inline framework/src/media/FileOutputDataSource.cpp]: OK instead of 0, sometimes 0 means negative for another case.

**Comment by seinfra** [comment]
Target : [12bad6051e80c93476fe209ea4388f06731b6128] - Code Rule Check (C++) OK.

File: framework/src/media/FileOutputDataSource.cpp (hunk 1 above)
Comment (junmin-kim) [reply framework/src/media/FileOutputDataSource.cpp]: Instead of `fclose() == 0`, I modified it to `fclose() != EOF`

**Comment by seinfra** [comment]
Target : [17ecb4271acfa450acc57c918fc2f3fc3a76b97d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [17ecb4271acfa450acc57c918fc2f3fc3a76b97d] - Code Rule Check Result:
framework/include/media/DataSource.h:66: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/DataSource.h:66: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/FileInputDataSource.h:33: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/include/media/FileInputDataSource.h:54: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/FileInputDataSource.h:54: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:60: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:66: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:66: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/FileInputDataSource.h:66: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:113: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/FileOutputDataSource.h:32: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/include/media/FileOutputDataSource.h:48: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/FileOutputDataSource.h:48: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileOutputDataSource.h:58: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/FileOutputDataSource.h:58: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileOutputDataSource.h:64: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileOutputDataSource.h:70: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileOutputDataSource.h:70: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/FileOutputDataSource.h:70: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileOutputDataSource.h:111: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)

File: framework/src/media/FileInputDataSource.cpp [hunk 2]
```diff
@@ -83,12 +83,14 @@ bool FileInputDataSource::open()
 	return isPrepare();
 }
 
-void FileInputDataSource::close()
+bool FileInputDataSource::close()
 {
-	if (mFp) {
-		fclose(mFp);
+	if (mFp && fclose(mFp) != EOF) {
```
Comment (Taejun-Kwon) [inline framework/src/media/FileInputDataSource.cpp]: Good, EOF is more good option

**Pull request merged**